app/scripts/controllers/provider-approval.js

```diff
@@ -6,27 +6,23 @@ const createAsyncMiddleware = require('json-rpc-engine/src/createAsyncMiddleware
  * A controller that services user-approved requests for a full Ethereum provider API
  */
 class ProviderApprovalController extends SafeEventEmitter {
-  /**
-   * Determines if caching is enabled
-   */
-  caching = true
-
   /**
    * Creates a ProviderApprovalController
    *
    * @param {Object} [config] - Options to configure controller
    */
-  constructor ({ closePopup, keyringController, openPopup, preferencesController } = {}) {
+  constructor ({ closePopup, initState, keyringController, openPopup, preferencesController } = {}) {
     super()
     this.closePopup = closePopup
     this.keyringController = keyringController
     this.openPopup = openPopup
     this.preferencesController = preferencesController
-    this.store = new ObservableStore({
-      approvedOrigins: {},
-      dismissedOrigins: {},
+    this.memStore = new ObservableStore({
```

Comment:
Do we definitely need the distinct `store` and `memStore`?

Reply:
This is the easiest way to persist one store and not the other

Reply:
oh, right, the memstore is not persisted, I was momentarily confused on that